stakingModel/contracts/InterestDistribution.sol

```diff
@@ -0,0 +1,193 @@
+pragma solidity 0.5.4;
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+
+library InterestDistribution {
+    using SafeMath for uint256;
+
+    /**
+     * @dev Structure to store Interest details.
+     * It contains total amount of tokens staked and total amount of interest generated.
+     */    
+    struct InterestData {
+      uint grandTotalStaked;
+      uint interestAccrued;
+      uint accumulatedYieldPerToken;
+      mapping(address => Staker) stakers;
+    }
+
+    /**
+     * @dev Structure to store staking details.
+     * It contains amount of tokens staked and blocknumber at which last staked.
+     */
+    struct Staker {
+        uint256 stakedToken;
+        uint256 weightedStake; // stored with 2 precision points ie., 0.82 => 82
+        uint256 lastStake;
+        uint256 withdrawnToDate;
+        uint256 avgYieldRatePerToken;
+    }
+
+    // 10^18
+    uint256 constant DECIMAL1e18 = 10 ** 18;
+
+    /**
+      * @dev Updates InterestData and Staker data while staking.
+      * 
+      * @param _interestData           Interest data
+      * @param _staker                 Staker's address
+      * @param _stake                  Amount of stake
+      * @param _donationPer            Percentage will to donate.
+      * @param _dailyIntrest           Withdrawn interest by individual staker so far.
```

Comment:
withdrawn inteerest? description doesnt seem right

Reply:
Updated in recent push